File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Practicas/javaNapakalaki/NapakalakiGame/NumericBadConsequence.java
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author manjaro
 */

public class NumericBadConsequence extends BadConsequence  {
    private int nVisibleTreasures;
    private int nHiddenTreasures;

    public NumericBadConsequence(String text, int levels , int nVisible, int nHidden) {

        super(text, levels);
        this.nVisibleTreasures = nVisible;
        this.nHiddenTreasures = nHidden;
    }

    @Override
    public boolean isDeath() {
        return false;
    }

    @Override
    public void substractVisibleTreasure(Treasure t){}

    @Override
    public void substractHiddenTreasure(Treasure t){}

    @Override
    public void setnVisibleTreasures(int n) {
        nVisibleTreasures = n;
    }

    @Override
    public void setnHiddenTreasures(int n) {
        nHiddenTreasures = n;
    }

    @Override
    public int getnVisibleTreasures() {
        return nVisibleTreasures;
    }

    @Override
    public int getnHiddenTreasures() {
        return nHiddenTreasures;
    }

    @Override
    public boolean isEmpty(){
       boolean res = false;
       if(super.getText() == "" && super.getLevels() == 0 && nVisibleTreasures == 0 && nHiddenTreasures == 0 ){
          res = true;
       }
       return res;
    }

    @Override
    public  BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v , ArrayList<Treasure> h){

      BadConsequence res;
      int visibles = 0;
      int ocultos = 0;
      int tamV = v.size();
      int tamH = h.size();

      if(nVisibleTreasures <= tamV ){
          visibles = nVisibleTreasures;

      }else if(tamV < nVisibleTreasures){
          visibles = tamV;
      }

      if(nHiddenTreasures <= tamH ){
          ocultos = nHiddenTreasures;
      }else if(tamH < nHiddenTreasures){
         ocultos = tamH;
      }
      res =  new NumericBadConsequence("", super.getLevels(), visibles, ocultos);
    return res;
 }

    @Override
     public String toString(){

        String res = super.getText() + "\nPierdes: " + super.getLevels() + " niveles "
                +  "\n" + Integer.toString(nVisibleTreasures)+" tesoros visibles\n"
                + Integer.toString(nHiddenTreasures) + " tesoros ocultos ";
      return res;
      }

     public String toStringPendingVisible(){
         String res = Integer.toString(nVisibleTreasures);
         return res;
     }

     public String toStringPendingOculto(){
         String res = Integer.toString(nHiddenTreasures);
         return res;
     }

}

File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Practicas/rubyNapakalaki/numeric_bad_consequence.rb

class NumericBadConsequence < Bad_consequence


    attr_accessor :nVisibleTreasures , :nHiddenTreasures

  def initialize(text, levels , nVisible, nHidden)
     super(text, levels)
     @nVisibleTreasures = nVisible
     @nHiddenTreasures = nHidden

  end


    def is_dead
      res = false

      res
    end

    def substract_visible_treasure(t)
      @nVisibleTreasures = @nVisibleTreasures-1
    end

   def substract_hidden_treasure(t)

       @nHiddenTreasures = @nHiddenTreasures -1
   end

    def setnVisibleTreasures(n)

        @nVisibleTreasures = n
    end

    def setnHiddenTreasures(n)

        @nHiddenTreasures = n
    end

    def getnVisibleTreasures
       @nVisibleTreasures
    end

    def getnHiddenTreasures
        @nHiddenTreasures
    end

    def is_empty
       res = false
         if @text == "" and @levels == 0 and @nVisibleTreasures == 0 and @nHiddenTreasures == 0
                    res = true
         end

       res

    end

    def adjust_to_fit_treasure_list(v, h)

         visibles=0
         ocultos=0
         tam_v = v.length
         tam_h = h.length

         if @nVisibleTreasures <= tam_v
               visibles = @nVisibleTreasures

         else

            if tam_v < @nVisibleTreasures
                          visibles = tam_v
           end
         end

         if @nHiddenTreasures <= tam_h
                        ocultos = @nHiddenTreasures

         else

            if tam_h < @nHiddenTreasures
                       ocultos = tam_h
            end
         end

          res = NumericBadConsequence.new("",getLevels,visibles,ocultos)

    res
   end

    def to_s

      "#{@text} Pierdes: #{@levels} niveles #{@nVisibleTreasures} tesoros visibles y #{@nHiddenTreasures} tesoros ocultos"
    end

end
